queries.sql

```diff
@@ -16,4 +16,21 @@ SELECT * FROM animals WHERE neutered;
 -- Find all animals not named Gabumon.
 SELECT * FROM animals WHERE name != 'Gabumon';
 -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
-SELECT* FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT* FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+                  -- Milestone 2
+-- How many animals are there?
+SELECT count(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT count(*) FROM animals WHERE escape_attempts = 0 GROUP BY escape_attempts;
+-- Check the names and escape_attempts if the query was correct( extra )
+SELECT name, escape_attempts FROM animals WHERE escape_attempts= 0 GROUP BY name,escape_attempts;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS "Average Weight" from animals;
+-- Who escapes the most, neutered or not neutered animals?
+select neutered, avg(escape_attempts) as "Avarage Escape Attempts" from animals group by neutered;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT min(weight_kg), max(weight_kg) from animals;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT avg(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31';
```

Comment:
* I think you can select animals grouped by species for this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?)  for example this query targets a particular animal type from the animal table where the date of birth is between 1997 to 2021 `SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;  WHERE  date_of_birth >= '1997/01/01' AND date_of_birth <= '2021/31/12'`
* This is because the species column is used to specify the animal type 
* For this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?) you should add `SELECT species and then group by species` to your answer this way your query is targeting a specific animal type whose birthdate falls between `1990 to 2000`  
* Use this hint to answer all queries related to animal type